### DISCUSSION TITLE
Support new units names

### DISCUSSION THREAD

**Discussion by mdicss** [madrisan/nagios-plugins-linux#120] [Ideas]
Hi

With Grafana, Influx and Icinga, there is a problem with the difference between binary units and the new 'decimal units'. MB, MiB etc, see:
 https://github.com/Icinga/icinga2/issues/9486
 https://github.com/Icinga/icinga2/pull/7871
 https://github.com/Icinga/icinga2/issues/6849
For check_disk, there is a kind of a solution, supporting new units names, see check_disk in https://github.com/nagios-plugins/nagios-plugins/blob/master/plugins/check_disk.c
Please add a analogous change to check_memory.

Regards
Matthias

**Comment by madrisan** [comment]
Hi. Actually `check_disc.c` permit to output the values also in the kiB/MiB/GiB units, but kB/MB/GB are still calculated as their respective binary units due to backward compatibility issues.
Is this the behaviour you expect?

See the [PR#121](https://github.com/madrisan/nagios-plugins-linux/pull/121).

If you find this project useful, do not forget to add a star to let me known this ;)

**Reply by mdicss** [reply]
Yes, the same behaviour as in check_disk is a good solution.
It was a bad idea to change the meaning of MB, GB etc. eg. in Grafana. For now, the best way to prevent problems is to use MiB, GiB to indicate that binary measures are used.
Thank you for implementing this.

**Reply by madrisan** [reply]
Ok, fine. Will be available in version 31 planned very soon (28/08).

**Reply by mdicss** [reply]
Hi Davide
I'm testing check_memory now. There is a problem with the in some way conflicting options -b,-k,-m,-g and -u. When -m is used, it overrides a setting of -u MiB and analogous with the other options. I think, -u should override the other options.
Regards, Matthias

**Comment by madrisan** [comment]
The current behaviour seems to me coherent. The last command-line option is the winner one.
```
./plugins/check_memory -u MiB -m   -->   MB
./plugins/check_memory -m -u MiB   -->   MiB
```
Why should the `-u` option be more powerful?
I can possibly generate a usage error if the two modes are mixed but I am not convinced it's necessary.

**Reply by mdicss** [reply]
Ok, thank you. I did obviously did not test the order of the options. I think a remark in the help would do it.
